File: WebApplication/src/modules/user/models/user-search-result.dto.ts
import { AuthorityDto } from "../../nomenclature/dtos/authority.dto";
import { UserStatus } from "../enums/user-status.enum";
import { RoleDto } from "./role.dto";

export class UserSearchResultDto {
  id: number;
  fullname: string;
  phone: string;
  email: string;
  role: RoleDto;
  authority: AuthorityDto;
  status: UserStatus;
}

File: Zopoesht.Application/Users/Dtos/UserSearchResultDto.cs
using Zopoesht.Application.Nomenclatures.Dtos;
using Zopoesht.Data.Users.Enums;

namespace Zopoesht.Application.Users.Dtos
{
    public class UserSearchResultDto
    {
        public int Id { get; set; }

        public string Fullname { get; set; }

        public string Phone { get; set; }

        public string Email { get; set; }

        public RoleDto Role { get; set; }

        public AuthorityDto Authority { get; set; }

        public UserStatus Status { get; set; }
    }
}
